internal/flow/promote.go

```diff
@@ -115,8 +116,12 @@ func (PromoteEvent) Type() string {
 	return "promote"
 }
 
-func (p PromoteEvent) Body() interface{} {
-	return p
+func (p PromoteEvent) Marshal() ([]byte, error) {
+	return json.Marshal(p)
+}
+
+func (p *PromoteEvent) Unmarshal(data []byte) error {
+	return json.Unmarshal(data, p)
 }
```

Comment:
Won't this get duplicated quite a lot? 🤔

Reply:
Yes. I think some sort of serializer layer would be nice. Like go-swagger har different producers maybe. 

But for the four events for now, it seemed ok.

It’s also what we do in our internal event package actually, now that I think of it.

Reply:
Alright, lets just continue, but i'd like to keep it on the consideration list 👍

Reply:
I've added it to the todo list. 👍